operator/pkg/ingress/helpers.go

```diff
@@ -8,17 +8,29 @@ import (
 	"fmt"
 	"strconv"
 
+	"github.com/sirupsen/logrus"
 	networkingv1 "k8s.io/api/networking/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func isCiliumDefaultIngressController(ctx context.Context, c client.Client) bool {
+func isCiliumManagedIngress(ctx context.Context, c client.Client, logger logrus.FieldLogger, ing networkingv1.Ingress) bool {
+	ingressClassName := ingressClassName(ing)
+
+	if ingressClassName != nil && *ingressClassName == ciliumIngressClassName {
+		return true
+	}
+
+	// Check for default Ingress class
+	return (ingressClassName == nil || *ingressClassName == "") && isCiliumDefaultIngressController(ctx, c, logger)
+}
+
+func isCiliumDefaultIngressController(ctx context.Context, c client.Client, logger logrus.FieldLogger) bool {
 	ciliumIngressClass := &networkingv1.IngressClass{}
 	if err := c.Get(ctx, types.NamespacedName{Name: ciliumIngressClassName}, ciliumIngressClass); err != nil {
 		if !errors.IsNotFound(err) {
-			log.WithError(err).Warn("Failed to load Cilium IngressClass")
+			logger.WithError(err).Warn("Failed to load Cilium IngressClass")
 			return false
```

Comment:
in any case, the result is false, so probably we don't need to check for the error type

Reply:
This is just to differentiate between an unexpected technical error (that should be logged) and the absence of the `IngressClass` `cilium` (that we shouldn't log - as this will spam the log)

Reply:
but yes i can deleted the `return false` in the block

Reply:
deleted the "duplicated" `return false`